### PR TITLE
Wrap wiggle

### LINKED ISSUE #1114: Fill positive areas in ploting (add GMT wiggle module)
Hi,
I'm new to pygmt and wondering how to fill the positive areas in plotting?
Please find the figure below for more information:
fill
Thanks for your attention.
Best Regards,
Zhendong

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

Wrap [wiggle](https://docs.generic-mapping-tools.org/latest/wiggle.html).

The following special alias is added:
- [D](https://docs.generic-mapping-tools.org/latest/wiggle.html#d): `position`
- [T](https://docs.generic-mapping-tools.org/latest/wiggle.html#t): `track`
- [Z](https://docs.generic-mapping-tools.org/latest/wiggle.html#z): `scale`

A simple gallery example revised from [GMT examples](https://docs.generic-mapping-tools.org/latest/wiggle.html#examples) is added.

A simple test revised from the example above is added.

---

**To be updated:**

Some common alias needs to be added, e.g., [t](https://docs.generic-mapping-tools.org/latest/gmt.html#t-full) (`transparency`). Note that `column` will be renamed to `incols`.

Some special alias needs to be added, e.g., [A](https://docs.generic-mapping-tools.org/latest/wiggle.html#a), [C](https://docs.generic-mapping-tools.org/latest/wiggle.html#c), [F](https://docs.generic-mapping-tools.org/latest/wiggle.html#f), [I](https://docs.generic-mapping-tools.org/latest/wiggle.html#i)

---

Fixes #1114

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/wiggle.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    Z="scale",
+    B="frame",
+    D="position",
+    G="color",
+    T="track",
+    U="timestamp",
+    V="verbose",
+    W="pen",
+    X="xshift",
+    Y="yshift",
```
Comment (weiji14) [inline pygmt/src/wiggle.py:21]: Sort alphabetically.

```suggestion
    B="frame",
    D="position",
    G="color",
    J="projection",
    R="region",
    T="track",
    U="timestamp",
    V="verbose",
    W="pen",
    X="xshift",
    Y="yshift",
    Z="scale",
```

File: pygmt/src/wiggle.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {c}
+    columns : str or 1d array
+        Choose which columns are x, y, and z, respectively if input is provided
+        via *data*. E.g. ``columns = [0, 1, 2]`` or ``columns = "0,1,2"`` if
+        the *x* values are stored in the first column, *y* values in the second
+        one and *z* values in the third one. Note: zero-based indexing is used.
+    {p}
+    {t}
+        *transparency* can also be a 1d array to set varying transparency
+        for symbols.
```
Comment (weiji14) [inline pygmt/src/wiggle.py:80]: Passing in 1d array to `transparency` doesn't seem to be implemented in your code, are you planning to add it?

**Review by weiji14 — COMMENTED** [review]
Hi @core-man, just reviewed your implementation of `wiggle` and it looks pretty good! Only a few minor comments for now. Let us know if you need help with adding a unit test, you could just adapt your gallery example into a unit test (but simplify it as much as possible, take out unnecessary arguments like `frame` perhaps).

Oh, and just to prevent this PR from taking up too much time (speaking from experience), please don't wrap every single alias for `wiggle` right now, just keep it to the bare minimum. We can complete the long alias list in a separate PR.

File: pygmt/src/wiggle.py (hunk 2 above)
Comment (core-man) [reply pygmt/src/wiggle.py:80]: Good suggestion. Will try it.

**Comment by weiji14** [comment]
Oh, and please remember to add `wiggle` to `doc/api/index.rst`.

**Comment by core-man** [comment]
> Hi @core-man, just reviewed your implementation of `wiggle` and it looks pretty good! Only a few minor comments for now. Let us know if you need help with adding a unit test, you could just adapt your gallery example into a unit test (but simplify it as much as possible, take out unnecessary arguments like `frame` perhaps).

Thank you. Will Ping you when I need help. I am preparing for the job interview these two weeks. Will come back soon next week.
 
> Oh, and just to prevent this PR from taking up too much time (speaking from experience), please don't wrap every single alias for `wiggle` right now, just keep it to the bare minimum. We can complete the long alias list in a separate PR.

Good to know. I thought to wrap all the arguments in one PR.

File: pygmt/src/wiggle.py (hunk 2 above)
Comment (weiji14) [reply pygmt/src/wiggle.py:80]: To be honest, you don't have to add it in this PR. You can also remove these two lines and add it back in a separate PR. Saves you from having to write a unit test for it now :slightly_smiling_face:

Comment (core-man) [reply pygmt/src/wiggle.py:80]: Thanks. Then I will give it a low priority.

**Comment by seisman** [comment]
@core-man The dvc-diff job fails. Did you forget to run `dvc push` to upload the baseline images?

**Comment by core-man** [comment]
> @core-man The dvc-diff job fails. Did you forget to run `dvc push` to upload the baseline images?

I think I ran it so I am a little confused about this erro. I did not have such error when I first used it in #1065.

**Comment by seisman** [comment]
OK, the error is `Error: Resource not accessible by integration`. It's because you're using your own fork, so you can't access the PyGMT token, which means the dvc-diff workflow can't comment in this PR.

Not a big issue.

File: pygmt/src/wiggle.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    scale : str or float
+        Gives anomaly scale in data-units/distance-unit. Append **c**, **i**,
+        or **p** to indicate the distance unit (cm, inch, or point); if no unit
+        is given we use the default unit that is controlled by
+        :gmt-docs:`PROJ_LENGTH_UNIT <gmt.conf.html#term-PROJ_LENGTH_UNIT>`.
+    {B}
+    position : str
+        [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
+        **+w**\ *length*\ [**+j**\ *justify*]\ [**+al**\ |\ **r**]\
+        [**+o**\ *dx*\ [/*dy*]][**+l**\ [*label*]]
```
Comment (seisman) [inline pygmt/src/wiggle.py:60]:
```suggestion
        [**+o**\ *dx*\ [/*dy*]][**+l**\ [*label*]].
```

File: pygmt/src/wiggle.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        Either a data file name or a 2d numpy array with the tabular data.
+        Use parameter ``columns`` to choose which columns are x, y, z,
+        respectively.
+    {J}
+    {R}
+    scale : str or float
+        Gives anomaly scale in data-units/distance-unit. Append **c**, **i**,
+        or **p** to indicate the distance unit (cm, inch, or point); if no unit
+        is given we use the default unit that is controlled by
+        :gmt-docs:`PROJ_LENGTH_UNIT <gmt.conf.html#term-PROJ_LENGTH_UNIT>`.
```
Comment (seisman) [inline pygmt/src/wiggle.py:55]:
```suggestion
        :gmt-term:`PROJ_LENGTH_UNIT`.
```

File: pygmt/src/wiggle.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    {B}
+    position : str
+        [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
+        **+w**\ *length*\ [**+j**\ *justify*]\ [**+al**\ |\ **r**]\
+        [**+o**\ *dx*\ [/*dy*]][**+l**\ [*label*]]
+        Defines the reference point on the map for the vertical scale bar.
+    {G}
+    track : str
+        Draw track [Default is no track]. Append pen attributes to use
+        [Defaults: width = 0.25p, color = black, style = solid].
```
Comment (seisman) [inline pygmt/src/wiggle.py:65]:
```suggestion
        [Default is 0.25p,black,solid].
```

File: pygmt/src/wiggle.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        or **p** to indicate the distance unit (cm, inch, or point); if no unit
+        is given we use the default unit that is controlled by
+        :gmt-docs:`PROJ_LENGTH_UNIT <gmt.conf.html#term-PROJ_LENGTH_UNIT>`.
+    {B}
+    position : str
+        [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
+        **+w**\ *length*\ [**+j**\ *justify*]\ [**+al**\ |\ **r**]\
+        [**+o**\ *dx*\ [/*dy*]][**+l**\ [*label*]]
+        Defines the reference point on the map for the vertical scale bar.
+    {G}
```
Comment (seisman) [inline pygmt/src/wiggle.py:62]: The `color` parameter needs more explanation.

File: pygmt/src/wiggle.py (hunk 5 above)
Comment (core-man) [reply pygmt/src/wiggle.py:65]: Shall we use **0.25p,black,solid** as suggested by [Example code stardard](https://github.com/GenericMappingTools/pygmt/blob/master/CONTRIBUTING.md#example-code-standards)?
```
*0.25p,black,solid*
```

File: pygmt/src/wiggle.py (hunk 6 above)
Comment (core-man) [reply pygmt/src/wiggle.py]: Done in 5e6c4bc.

**Comment by core-man** [comment]
We sort alias by alphabetically. How about the descriptions of those parameters, alphabetically or similar to the GMT order? first pygmt parameters then GMT alias? I prefer the GMT order.

**Comment by seisman** [comment]
> We sort alias by alphabetically. How about the descriptions of those parameters, alphabetically or similar to the GMT order? first pygmt parameters then GMT alias? I prefer the GMT order.

We don't have a rule for the parameter orders now, but see #884 for some discussions. Personally, I prefer to sort parameter alphabetically rather than following the GMT order.

File: pygmt/src/wiggle.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    Full parameter list at :gmt-docs:`wiggle.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    x/y/z : float or 1d arrays
+        The x and y coordinates, or arrays of x and y coordinates of the
+        z data point or an array of z data points.
```
Comment (seisman) [inline pygmt/src/wiggle.py:44]: I don't test it, but does a single data point (`float`) makes sense for `wiggle`?

File: pygmt/src/wiggle.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    data : str or 2d array
+        Either a data file name or a 2d numpy array with the tabular data.
+        Use parameter ``columns`` to choose which columns are x, y, z,
+        respectively.
```
Comment (seisman) [inline pygmt/src/wiggle.py:48]: Just a note here, we'll rename it to `incols` instead.

File: pygmt/src/wiggle.py (hunk 7 above)
Comment (core-man) [reply pygmt/src/wiggle.py]: Yes, you're right. a single data point input means nothing is plotted. Revised in 0880391.

File: pygmt/src/wiggle.py (hunk 8 above)
Comment (core-man) [reply pygmt/src/wiggle.py:47]: Noted.

File: examples/gallery/seismology/wiggle.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
+"""
+Wiggle along tracks
+--------------------
+
+The :meth:`pygmt.Figure.wiggle` method can plot z = f(x,y) anomalies along
+tracks. ``x``, ``y``, ``z`` can be specified as 1d arrays or within a specified
+file. The ``scale`` parameter can be used to set anomaly scale in
```
Comment (michaelgrund) [inline examples/gallery/seismology/wiggle.py:7]:
```suggestion
file. The ``scale`` parameter can be used to set the anomaly scale in
```

File: examples/gallery/seismology/wiggle.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    x=x,
+    y=y,
+    z=z,
+    # Set anomaly scale to "20c"
+    scale="20c",
+    # Fill positive and negative areas red and gray, respectively
+    color=["red+p", "gray+n"],
+    # Set the outline width to "1.0p"
+    pen="1.0p",
+    # Draw a blue track of width 0.5 points
```
Comment (michaelgrund) [inline examples/gallery/seismology/wiggle.py:32]:
```suggestion
    # Draw a blue track with a width of 0.5 points
```

File: examples/gallery/seismology/wiggle.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    track="0.5p,blue",
+    # Plot a vertical scale bar at the right middle. The bar length is 100 in
+    # data (z) units. Set the z unit lable to "nT".
```
Comment (michaelgrund) [inline examples/gallery/seismology/wiggle.py:35]:
```suggestion
    # data (z) units. Set the z unit label to "nT".
```

**Comment by core-man** [comment]
@michaelgrund Thanks for the review.

File: examples/gallery/seismology/wiggle.py [hunk 12]
```diff
@@ -0,0 +1,38 @@
+"""
+Wiggle along tracks
+--------------------
```
Comment (weiji14) [inline examples/gallery/seismology/wiggle.py:3]:
```suggestion
Wiggle along tracks
-------------------
```

File: examples/gallery/seismology/wiggle.py [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+The :meth:`pygmt.Figure.wiggle` method can plot z = f(x,y) anomalies along
+tracks. ``x``, ``y``, ``z`` can be specified as 1d arrays or within a specified
+file. The ``scale`` parameter can be used to set the anomaly scale in
+data-units/distance-unit. The positive and/or negative areas can be filled with
```
Comment (weiji14) [inline examples/gallery/seismology/wiggle.py:8]:
```suggestion
file. The ``scale`` parameter can be used to set the scale of the anomaly in
data/distance units. The positive and/or negative areas can be filled with
```

File: pygmt/tests/test_wiggle.py [hunk 14]
```diff
@@ -0,0 +1,32 @@
+# pylint: disable=redefined-outer-name
```
Comment (weiji14) [inline pygmt/tests/test_wiggle.py:1]: Is this pylint disable line needed?
```suggestion
```

**Review by weiji14 — COMMENTED** [review]
I'm happy with the `def wiggle()` code implementation and single unit test, and you've wrapped the main required parameter scale (Z) and a couple of other basic ones which is good. Just a few minor edits and it should be :crossed_fingers: close to merging.

**Comment by weiji14** [comment]
This is an auto-generated report of images that have changed on the DVC remote
  
  | Status | Path |
  |----------|--------------------------------------|
  | added | pygmt/tests/baseline/test_wiggle.png |
  
  ## Image diff(s)
  
  <details>
  
  ### Added images
  
  - pygmt/tests/baseline/test_wiggle.png 
  
  
  ### Modified images
  
  | Path | Old | New |
  |---|---|---|
  </details>

**Review by weiji14 — APPROVED** [review]

File: pygmt/src/wiggle.py [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        or **p** to indicate the distance unit (cm, inch, or point); if no unit
+        is given we use the default unit that is controlled by
+        :gmt-term:`PROJ_LENGTH_UNIT`.
+    {B}
+    position : str
+        [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
+        **+w**\ *length*\ [**+j**\ *justify*]\ [**+al**\ |\ **r**]\
+        [**+o**\ *dx*\ [/*dy*]][**+l**\ [*label*]].
+        Defines the reference point on the map for the vertical scale bar.
+    color: str
```
Comment (seisman) [inline pygmt/src/wiggle.py:61]:
```suggestion
    color : str
```

File: pygmt/src/wiggle.py [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+        Set fill shade, color or pattern for positive and/or negative wiggles
+        [Default is no fill]. Optionally, append **+p** to fill positive areas
+        (this is the default behavior). Append **+n** to fill negative areas.
+        Append **+n+p** to fill both positive and negative areas with the same
+        fill. Note: You will need to repeat the color parameter to select
+        different fills for the positive and negative wiggles.
+
+    track : str
+        Draw track [Default is no track]. Append pen attributes to use
+        [Default is **0.25p,black,solid**\ ].
```
Comment (seisman) [inline pygmt/src/wiggle.py:71]:
```suggestion
        [Default is **0.25p,black,solid**].
```

File: examples/gallery/seismology/wiggle.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
+"""
```
Comment (seisman) [inline examples/gallery/seismology/wiggle.py:1]: I don't think `wiggle` is a seismology-specific module. I recommend moving the example to the "Line" category.

**Review by michaelgrund — APPROVED** [review]
Agree with @seisman to move the gallery example to the "Line" category since e.g. traces of active reflection seismic data are also often plotted in such way (and not only seismology related content). Except, I'm fine with that PR. Great work!

File: examples/gallery/seismology/wiggle.py (hunk 17 above)
Comment (core-man) [reply examples/gallery/seismology/wiggle.py:1]: okay.

**Pull request merged**